src/settings_internal.h

```diff
@@ -12,21 +12,20 @@
 
 #include "saveload/saveload.h"
 
-enum SettingGuiFlag : uint16 {
-	/* 2 bytes allocated for a maximum of 16 flags. */
-	SGF_NONE = 0,
-	SGF_0ISDISABLED   = 1 << 0, ///< a value of zero means the feature is disabled
-	SGF_DISPLAY_ABS   = 1 << 1, ///< display absolute value of the setting
-	SGF_MULTISTRING   = 1 << 2, ///< the value represents a limited number of string-options (internally integer)
-	SGF_NETWORK_ONLY  = 1 << 3, ///< this setting only applies to network games
-	SGF_CURRENCY      = 1 << 4, ///< the number represents money, so when reading value multiply by exchange rate
-	SGF_NO_NETWORK    = 1 << 5, ///< this setting does not apply to network games; it may not be changed during the game
-	SGF_NEWGAME_ONLY  = 1 << 6, ///< this setting cannot be changed in a game
-	SGF_SCENEDIT_TOO  = 1 << 7, ///< this setting can be changed in the scenario editor (only makes sense when SGF_NEWGAME_ONLY is set)
-	SGF_PER_COMPANY   = 1 << 8, ///< this setting can be different for each company (saved in company struct)
-	SGF_SCENEDIT_ONLY = 1 << 9, ///< this setting can only be changed in the scenario editor
+enum SettingFlag : uint16 {
+	SF_NONE = 0,
+	SF_GUI_0_IS_SPECIAL  = 1 << 0, ///< a value of zero is possible and has a custom string.
+	SF_GUI_DISPLAY_ABS   = 1 << 1, ///< display absolute value of the setting.
+	SF_GUI_MULTISTRING   = 1 << 2, ///< the value represents a limited number of string-options (internally integer).
+	SF_GUI_CURRENCY      = 1 << 3, ///< the number represents money, so when reading value multiply by exchange rate.
+	SF_NETWORK_ONLY      = 1 << 4, ///< this setting only applies to network games.
+	SF_NO_NETWORK        = 1 << 5, ///< this setting does not apply to network games; it may not be changed during the game.
+	SF_NEWGAME_ONLY      = 1 << 6, ///< this setting cannot be changed in a game.
+	SF_SCENEDIT_TOO      = 1 << 7, ///< this setting can be changed in the scenario editor (only makes sense when SF_NEWGAME_ONLY is set).
+	SF_SCENEDIT_ONLY     = 1 << 8, ///< this setting can only be changed in the scenario editor.
+	SF_PER_COMPANY       = 1 << 9, ///< this setting can be different for each company (saved in company struct).
```

Comment:
Is reordering this going to cause any issues?

Reply:
not as far as I can see and as far as I could test.

Nowhere this value is stored in the savegame. But I would love some extra pairs of eyes on it to validate that claim.

Reply:
Shouldn't the comments start with a capital?